### PR TITLE
add DescribeWorkflowExecution

### PR DESCRIPTION
**Pull request by yiminc**
changes:
1) added DescribeWorkflowExecution to idl files (both frontend and history) with its supporting types
2) frontend forward request to history which forward to historyEngine
3) history engine load mutable state, and return result.
4) rename transfer task DeleteExecutionTask to CloseExecutionTask, change it so it won't remove the closed workflow execution from executions table
5) update timer task DeleteHistoryEventTask to remove workflow execution as well.

So now the mutable state will be available after execution is closed (before its retention date), and it will be removed along with history events by the delete history timer task.

I think we should also remove history's GetWorkflowExecutionNextEventID and replace it with this new DescribeWorkflowExecution because it contains super set of the fields and the cost is the same, they both needs to load the mutable state. But it can be in separate diff.

### CODE REVIEW

**Comment by mfateev** [comment]
Awesome. After this change implementing ability to fail workflow start on already closed workflow with the same ID is trivial. Please implement it while you are on it as I'm sure customers are going to ask for it very soon.

File: common/metrics/defs.go [hunk 1]
```diff
@@ -430,7 +439,7 @@ var ScopeDefs = map[ServiceIdx]map[int]scopeDefinition{
 		TransferQueueProcessorScope:                 {operation: "TransferQueueProcessor"},
 		TransferTaskActivityScope:                   {operation: "TransferTaskActivity"},
 		TransferTaskDecisionScope:                   {operation: "TransferTaskDecision"},
-		TransferTaskDeleteExecutionScope:            {operation: "TransferTaskDeleteExecution"},
+		TransferTaskCloseExecutionScope:             {operation: "TransferTaskCloseExecution"},
```
Comment (samarabbas) [inline common/metrics/defs.go]: Any specific reason to rename this?

File: common/metrics/defs.go [hunk 2]
```diff
@@ -418,7 +426,8 @@ var ScopeDefs = map[ServiceIdx]map[int]scopeDefinition{
 		HistoryRespondActivityTaskCompletedScope:    {operation: "RespondActivityTaskCompleted"},
 		HistoryRespondActivityTaskFailedScope:       {operation: "RespondActivityTaskFailed"},
 		HistoryRespondActivityTaskCanceledScope:     {operation: "RespondActivityTaskCanceled"},
-		HistoryGetWorkflowExecutionNextEventIDScope: {operation: "GetWorkflowExecutionNextEventIDScope"},
+		HistoryGetWorkflowExecutionNextEventIDScope: {operation: "GetWorkflowExecutionNextEventID"},
+		HistoryDescribeWorkflowExecutionScope:       {operation: "HistoryDescribeWorkflowExecution"},
```
Comment (samarabbas) [inline common/metrics/defs.go]: Other operation names are not prefixed with History.

File: common/metrics/defs.go (hunk 1 above)
Comment (yiminc) [reply common/metrics/defs.go]: this is no longer deleting the execution, it is just close it and copy data to visibility store. So I think the name with delete is confusing.

Comment (samarabbas) [reply common/metrics/defs.go]: Got it.

**Pull request merged**